### PR TITLE
Update filtering documentation

### PR DESCRIPTION
**Pull request by danielnelson**
I tried to clarify some of the issues surrounding tagpass, reordered taginclude/tagexclude to match the other filters, and tried to correct some of the terminology.

### Required for all PRs:

- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [ ] README.md updated (if adding a new plugin)

### CODE REVIEW

**Comment by danielnelson** [comment]
cc @phemmer

File: docs/CONFIGURATION.md [hunk 1]
```diff
@@ -124,31 +124,41 @@ is not specified then processor execution order will be random.
...
-* **taginclude**: taginclude is the inverse of tagexclude. It will only include
-the tag keys in the final measurement.
-
-**NOTE** `tagpass` and `tagdrop` parameters must be defined at the _end_ of
-the plugin definition, otherwise subsequent plugin config options will be
-interpreted as part of the tagpass/tagdrop map.
+* **namepass**:
+An array of strings that is used to filter metrics generated by the
+current input. Each string is tested as a glob match against measurement names
+and only if it matches, the field is emitted.
```
Comment (phemmer) [inline docs/CONFIGURATION.md]: > only if it matches, the ~~field~~ is emitted

the point is emitted

File: docs/CONFIGURATION.md [hunk 2]
```diff
@@ -124,31 +124,41 @@ is not specified then processor execution order will be random.
...
+* **namedrop**: The inverse of `namepass`.  If the measurement name matches it is
+discarded.
+* **fieldpass**:
+An array of strings that is used to filter metrics generated by the current
+input. Each string in the array is matched as a glob pattern against field
+names and if it matches, the field is emitted. `fieldpass` is not available
+for outputs.
```
Comment (phemmer) [inline docs/CONFIGURATION.md]: Should not that *only* fields which match are emitted. All other fields will be dropped.

File: docs/CONFIGURATION.md [hunk 3]
```diff
@@ -124,31 +124,41 @@ is not specified then processor execution order will be random.
...
+* **fielddrop**:
+The inverse of `fieldpass`, if a field key matches, it is not emitted.
+`fielddrop` is not available for outputs.
+* **tagpass**:
+A table mapping tag keys to arrays of tag value patterns.  Each pattern is
+matched as a glob against the corresponding tag key and if the point contains
```
Comment (phemmer) [inline docs/CONFIGURATION.md]: > the ~~corresponding~~ tag ~~key~~

the tag value

&nbsp;

And clarification why to omit "corresponding"

> Each pattern is matched ...

There is only one tag value. So each pattern corresponds to what?

File: docs/CONFIGURATION.md [hunk 4]
```diff
@@ -124,31 +124,41 @@ is not specified then processor execution order will be random.
...
+a matching tag value then the point is emitted.
+* **tagdrop**:
+The inverse of `tagpass`.  If a match is found, the point is not emitted. This
+is tested on points after they have passed the `tagpass` test.
+* **taginclude**:
+Used to whitelist tags on a point.  In contrast to `tagpass`, which will
+pass an entire point based on its tag, `taginclude` passes only the specified
+tags from the measurement.  This filter can be used on both inputs & outputs,
+but it is _recommended_ to be used on inputs, as it is more efficient to
+filter out tags at the ingestion point.
```
Comment (phemmer) [inline docs/CONFIGURATION.md]: The description doesn't cover how to format the config param. An array of tag names to glob. non-matching tag is dropped.

**Comment by danielnelson** [comment]
I rewrote it all again.  Made sure to start with the type of the field and tried to simplify/clarify.

File: docs/CONFIGURATION.md [hunk 5]
```diff
@@ -124,31 +124,40 @@ is not specified then processor execution order will be random.
...
-the plugin definition, otherwise subsequent plugin config options will be
-interpreted as part of the tagpass/tagdrop map.
+* **namepass**:
+An array of glob pattern strings.  Only points whose measurement name matches
+a pattern in this list are emitted.
+* **namedrop**:
+The inverse of `namepass`.  If a match is found the point is discarded. This
+is tested on points after they have passed the `namepass` test.
+* **fieldpass**:
+An array of glob pattern strings.  Only fields whose field key matches an
```
Comment (phemmer) [inline docs/CONFIGURATION.md]: Nit:

> matches ~~an~~

matches a

Comment (danielnelson) [inline docs/CONFIGURATION.md]: thanks

**Pull request merged**